torch/csrc/Module.cpp

```diff
@@ -452,7 +452,7 @@ PyObject *THPModule_setFlushDenormal(PyObject *_unused, PyObject *arg) {
 
 PyObject *THPModule_getDefaultDtype(PyObject *_unused, PyObject *arg) {
   HANDLE_TH_ERRORS
-  auto scalar_type = torch::tensors::get_default_scalar_type();
+  auto scalar_type = at::typeMetaToScalarType(torch::tensors::get_default_tensor_options().dtype());
```

Comment:
I don't think we should do this.

There is no such thing as default tensor options.  There's a default dtype, and a default device type for constructors, but that's it.  Using tensor options for this implies there are dimensions of the default which are absolutely not applicable: currently layout, optionality, etc.  But soon there will also be dimension names and that doesn't make sense in this context.

The right way to think about this, I think, is that TensorOptions are a mechanism for C++ users to specify kwarg-like options to our factory function.  Every other use of them has been a mistake (`Type.options()`, hacking up our parsing in multiple places to pass TensorOptions instead of just the actual parameters, etc.)

Reply:
Okay, I'll replace this with get_default_tensor_type_id, which is better than get_default_tensor_type because we are still having a default backend, but we just don't have scalar_type info passed along with it.

Reply:
In the future we should probably have a default device type instead.